File: src/main/java/net/vidageek/games/vraptor/GameFactory.java
package net.vidageek.games.vraptor;

import net.vidageek.games.Game;
import net.vidageek.games.regex.Descriptions;
import net.vidageek.games.regex.task.RegexGame;
import br.com.caelum.vraptor.ioc.Component;
import br.com.caelum.vraptor.ioc.ComponentFactory;

@Component
final public class GameFactory implements ComponentFactory<Game> {

	private final Descriptions descriptions;

	public GameFactory(final Descriptions descriptions) {
		this.descriptions = descriptions;
	}

	public Game getInstance() {
		return new RegexGame(descriptions);
	}

}

File: src/main/scala/net/vidageek/games/vraptor/GameFactory.scala
package net.vidageek.games.vraptor

import br.com.caelum.vraptor.ioc.Component
import br.com.caelum.vraptor.ioc.ComponentFactory
import net.vidageek.games.regex.task.RegexGame
import net.vidageek.games.regex.Descriptions
import net.vidageek.games.Game

@Component
class GameFactory(descriptions: Descriptions) extends ComponentFactory[Game] {

  def getInstance: Game = new RegexGame(descriptions)
}